projects/packages/my-jetpack/_inc/components/my-jetpack-tab-panel/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { TabPanel } from '@wordpress/components';
+import { useCallback } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { MY_JETPACK_SECTION_OVERVIEW } from './constants';
+import styles from './styles.module.scss';
+import { TabContent } from './tab-content';
+import { MyJetpackSection } from './types';
+import { getMyJetpackSections, isValidMyJetpackSection } from './utils';
+
+/**
+ * My Jetpack Tab panel component.
+ *
+ * @return The rendered component.
+ */
+export function MyJetpackTabPanel() {
+	const params = useParams();
+	const navigate = useNavigate();
+
+	const onTabSelect = useCallback(
+		( tabName: string ) => {
+			if ( tabName !== params.section ) {
+				navigate( `/${ tabName }` );
+			}
```

Comment:
should we have a guard here, to prevent setting random tab names (and blank page)?

or better, use [replaceState()](https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState) to keep the url in sync? otherwise `/blabla` will display "Overview" page, while not critical - it's not perfect as well.

Reply:
But since we control the tabs, where will the random value come from?

Reply:
I have fixed it in ae26423cc76ebabacbf738d15d00b13f0b03cd6b

You can see what happens if I update the URL manually.


https://github.com/user-attachments/assets/6ff224b5-2e8a-49de-9a3f-30319cc48a7f